docs/Configuration/Dialplan/Pattern-Matching.md

```diff
@@ -3,331 +3,255 @@ title: Pattern Matching
 
-X
--
+Character Set
+-------------
 
-The letter **X** or **x** represents a single digit from 0 to 9.
+Place the characters you want to match between square brackets. If you want to match an `a` or an `e`, use `[ae]`, like in `gr[ae]y`, which matches either `gray` or `grey`.
 
-Z
--
+**Examples**:
+* `_gr[ae]y`: Matches `gray`, `grey`
+* `_[1-468]`: Matches `1`, `2`, `3`, `4`, `6`, `8`. It does not match any number from one to four hundred sixty-eight!
+* `_a[c-e]`: Matches `ac`, `ad`, `ae`
+* `_[127-9ac-e]`: Matches `1`, `2`, `7`, `8`, `9`, `a`, `c`, `d`, `e`
 
-The letter **Z** or **z** represents any digit from 1 to 9.
 
-N
--
+Special Symbols
+---------------
 
-The letter **N** or **n** matches any digit from 2-9.
+Some Character Sets can be written in a short form. Here are the most common ones:
 
+* `_[0-9]` = `_X` (or `x`): Represents a single digit from 0 to 9.
+* `_[1-9]` = `_Z` (or `z`): Represents any digit from 1 to 9.
+* `_[2-9]` = `_N` (or `n`): Matches any digit from 2 to 9.
 
+**Examples**:
+* `_6[34]XX`: Short form of `_64XX` and `_63XX`, matches `6400`, `6300`, `6401`, `6301`, `6420`, `6320`, `6433`, `6333`, ...
+* `_6[34X]`: Matches `63`, `64`, `6X`. The `X`, `N`, and `Z` convenience notations have no special meaning within a set, they represent the literal character.
 
-Ranges used in Pattern Matching
-===============================
-
-A numeric range can be used to match against a dialed number. This is also called a Character Set
-
+!!! note
+    The only characters with special meaning within a set are the `-` character, to define a range between two characters, the  `\` character to escape a special character available within a set, and  
+[//]: # (end-note)
 
 
-[1237-9]
+Wilcards
 --------
 
-This pattern matches any digit or letter in the brackets. In this example, the pattern will match: 1,2,3,7,8,9
-
-
-
-Wilcards used in Pattern Matching
-=================================
-
-The following special characters are considered wildcards
+Within Asterisk patterns, we can also use other characters to represent a *family* of dial numbers. The following special characters are considered wildcards:
 
-* . The '.' character matches one or more characters
-* ! The '!' character matches zero or more characters immediately
+* `!`: Matches zero or more characters *immediately*. Equivalent to RegEx's `*` quantifier. The behaviour of this wildcard is 
+* `.`: Matches one or more characters. Equivalent to RegEx's `+` quantifier.
 
 
+**Examples**:
+* `_9876.`: Matches any number that began with `9876` and has at least one more character or digit, *e.g* `98760`, `9876a`, `9876xxx`...
 
 The exclamation mark wildcard (!), behaves specially and will be further explained below in 'Other Special Characters' below.
```

Comment:
It might make sense to mention this directly in line 48 where "! " is being defined, and not separately here.

Reply:
I tried to explain better the line 48. Please, have a look.